allennlp/commands/train.py

```diff
@@ -170,19 +172,30 @@ def datasets_from_params(params: Params) -> Dict[str, Iterable[Instance]]:
 
     return datasets
 
-def create_serialization_dir(params: Params, serialization_dir: str) -> None:
+def create_serialization_dir(params: Params, serialization_dir: str, recover: bool) -> None:
     """
     This function creates the serialization directory if it doesn't exist.  If it already exists,
     then it verifies that we're recovering from a training with an identical configuration.
 
     Parameters
     ----------
-    params: Params, required.
+    params: ``Params``
         A parameter object specifying an AllenNLP Experiment.
-    serialization_dir: str, required
+    serialization_dir: ``str``
         The directory in which to save results and logs.
+    recover: ``bool``
+        If ``True``, we will try to recover from an existing serialization directory, and crash if
+        the directory doesn't exist, or doesn't match the configuration we're given.
     """
     if os.path.exists(serialization_dir):
+        if serialization_dir == '/output':
```

Comment:
Instead of special casing only for beaker, you could write the check to fail if there's a non-empty output directory.

Reply:
The reason I prefer it this way is so that we catch more errors on the command line - if you get to this branch of code (in particular, [this line](https://github.com/allenai/allennlp/pull/925/files#diff-d4e8f335bf383d8d2b18287949003bbeR202)) from the command line, you did something on accident, and the behavior will not be what you expected.  Much better to crash when the user did something on accident than to continue silently, or with a warning.

If there's ever another tool that people are using allennlp with, we can consider supporting it.  But for now, beaker and the command line are the two places where people run allennlp, and this is the best way to catch errors for both of those cases.

Reply:
I prefer to build software that catches clear error cases, but remains open to being used in ways that may be hard to anticipate for the original author.

It might not have been an accident.  For example, our test cases previously used an empty output directory because it was convenient when getting a temporary directory.